src/constructs/aws/ServerSideWebsite.ts

```diff
@@ -372,7 +384,7 @@ export class ServerSideWebsite extends AwsConstruct {
     private createRequestFunction(): cloudfront.Function {
         let additionalCode = "";
 
-        if (this.configuration.redirectToMainDomain === true) {
+        if (this.configuration.redirectToMainDomain === true && this.domains?.length > 1) {
```

Comment:
This duplicates #321, correct?

I think this should be reverted from this PR to avoid conflicts and keep things on topic.